x509-cert/src/certificate.rs

```diff
@@ -212,3 +215,61 @@ impl<P: Profile> PemLabel for CertificateInner<P> {
 ///
 /// [RFC 6066]: https://datatracker.ietf.org/doc/html/rfc6066#section-10.1
 pub type PkiPath = Vec<Certificate>;
+
+#[cfg(feature = "pem")]
+impl<P: Profile> CertificateInner<P> {
+    /// Parse a chain of pem-encoded certificates from a slice.
+    ///
+    /// Returns the list of certificates.
+    pub fn load_pem_chain(mut input: &[u8]) -> Result<Vec<Self>, Error> {
+        fn find_boundary<T>(haystack: &[T], needle: &[T]) -> Option<usize>
+        where
+            for<'a> &'a [T]: PartialEq,
+        {
+            haystack
+                .windows(needle.len())
+                .position(|window| window == needle)
+        }
+
+        let mut certs = Vec::new();
+        let mut position: usize = 0;
+
+        let end_boundary = {
+            let mut out = Vec::new();
+            out.extend_from_slice(b"-----END ");
+            out.extend_from_slice(Self::PEM_LABEL.as_bytes());
+            out.extend_from_slice(b"-----");
+            out
+        };
```

Comment:
This seems a little gross for what's effectively a constant, just to avoid repeating the word "CERTIFICATE".

Reply:
fair